### ISSUE TITLE
System specfic QT styling can make FastFlix hard to use

### ISSUE THREAD

**Issue by cdgriffith** [cdgriffith/FastFlix#191]
As reported by Nick Bailey:

```
~/Packages/FastFlix$ QT_STYLE_OVERRIDE=Adapta python -m
fastflix

it says

QApplication: invalid style override 'Adapta' passed, ignoring it.
        Available styles: Windows, Fusion

Using either of those doesn't change the appearance, which is unchanged from
leaving out the environment variable all together. I wouldn't mind, but it
makes the application hard to use (white text on white for pull-down menu
boxes...) I thought I should report this at least. Maybe you already know
about it.

The version I'm running was cloned from the git repo a day or two ago, and
reports itself as v4.1.2. Running on Debian Bullseye, in a virtual environment
as per your instructions.
```
![d64b20bb-b169-4ba7-afd2-126a78a2bf38](https://user-images.githubusercontent.com/3275435/106011059-b9d83700-607f-11eb-9222-457c54a12c96.png)

**Comment by nickbailey** [comment]
Adwaita is the theme I tried (Adapta is a typo :) )  Same result though. Also Breeze. No other info, sorry. I don't really get any further away from the metal that Qt. I've no idea how KDE themes work.

**Comment by nickbailey** [comment]
I found what may be pertinent information here: https://stackoverflow.com/questions/52426437/how-to-instruct-pyside2-application-to-use-qt5ct-settings
Contrary to what's written there, PySide2 is in fact packaged for Debian Bullseye, but without a metapackage. I copied the current FastFlix installation in to a different directory, and removed venv/lib/PySide2. Then I installed the Debian version of PySide2 like this:
`apt install $(apt-cache pkgnames python3-pyside2.) pyside2-tools`
Now outside the venv, I see all available themes:
```
>>> from PySide2.QtWidgets import QStyleFactory
>>> print(QStyleFactory.keys())
['Adwaita-Dark', 'Adwaita', 'Breeze', 'Oxygen', 'QtCurve', 'Windows', 'Fusion']
```
but on within the venv, I see:
```
nick@polonius:~/Packages$ cp -a FastFlix/ FastFlix-no-PySide
nick@polonius:~/Packages$ cd FastFlix-no-PySide/
nick@polonius:~/Packages/FastFlix-no-PySide$ rm -rf venv/lib/python3.9/site-packages/PySide2
```
I then edited the activate script to change the venv directory and made a link from the old PySide2 to the system one `/usr/lib/python3/dist-packages/PySide2` This causes the theme to be used, but the pull-down menus are stiill white-on white.
![Screenshot_20210130_153818](https://user-images.githubusercontent.com/1101370/106360596-32a1f200-6311-11eb-82f2-200841afc44b.png)
Changing the application style to Adwita in the control panel makes the menus readable (just) although something sitll isn't quite right.
![Screenshot_20210130_154039](https://user-images.githubusercontent.com/1101370/106360669-a6dc9580-6311-11eb-9f32-c1511b8212c1.png)

**Comment by nickbailey** [comment]
@GitKike98's little rant about Flat UIs gave me a further clue. If I turn of FlatUI in the settings (and do all the stiff with the local PySide2 install)... presto!
![Screenshot_20210131_133206](https://user-images.githubusercontent.com/1101370/106385549-c1be1100-63c8-11eb-86cd-e846eea8afc9.png)

What's a FlatUI? (I've a feeling I should know this... )

BTW, well done for not being offended by GitKike98. From the G'Day up front, I guess he's Australian. I once got an SMS from a Professor of Pschology (Dean of Faculty no less) to check he'd got my number right and it said "What's your bloody story?" Our local Australian said that was really friendly. I was really surprised, and I stay in Glasgow!

Anyway, maybe you can mark this closed and maybe do a FAQ about switching FlatUI off, or maybe it's just a less urgent bug.

**Comment by cdgriffith** [comment]
Good discovery work!

Yes, Flat UI is just some custom QSS to flatten the buttons / combo boxes to make it look like a more modern UI. The problem is it doesn't redefine everything, so if the OS is providing custom colors and stuff to QT to use, looks to clash horribly so best to just use the default settings. 

So real solution might be to just redefine all the defaults that I expect to be there, so that it has a unified UI across systems.

**Comment by nickbailey** [comment]
I'm not clear why that can't be achieved with a flat desktop theme... but I'm old skool and don't really even like mice. It's been all down hill since gopher servers fell out of favour if you ask me ask me. So maybe it's a QSS bug and not a FastFlix one? In which case you could close it (unless you wrote the stylesheet ;) ) The bumpy one on the screeshot above is quite flat enough for me, so +1 happy user.

**Comment by cdgriffith** [comment]
yup that was totally a custom design style sheet, so on me. Leave it open for now. Not a something I will address quickly as there is a "fix" by unselecting the "Flat UI" option in settings, but here as a backlog item

**Comment by cdgriffith** [comment]
This should now be resolved with the `onyx` theme in 4.5.0

**Comment by cdgriffith** [comment]
Marking this as done as this is hopefully fixed and hasn't been a follow up. If there is still any issues please report back and can reopen. Thanks again for bringing it to my attention!
